src/types/config.ts

```diff
@@ -192,9 +192,15 @@ interface MicroservicesConfig {
 
 export interface NonEVMChain {
 	id: number;
+	particleChainId: number;
 	name: string;
 	chainType: ChainType;
 	adapterNetwork: WalletAdapterNetwork;
+	nativeCurrency: {
+		name: string;
+		symbol: string;
+		decimals: number;
+	};
```

Comment:
Do you know if we need to define them here? Actually the SOL token always has the id of Solana system program, look at this in `useCreateSolanaDonation.tsx`
```ts

		if (address !== SystemProgram.programId.toBase58()) {
			throw new Error(
				'Token address is not native token address - not supported yet!',
			);
		}
```

Reply:
I've used "nativeCurrency" in "CryptoDonation" L:283. I've removed below code from that component, so need "nativeCurrency" here.
```
if (walletChainType === ChainType.SOLANA) {
			return setSelectedTokenBalance(
				BigInt(Number(balance || 0) * LAMPORTS_PER_SOL),
			);
		}
```

My idea was to avoid adding chain related hard coded data as much as possible. If we use "SystemProgram.programId", first we need to check if it's Solana or not, and we should avoid adding these conditions or checking chains one by one. I think it's better to try to detect chain data (like logo, currency, chain name, ...) automatically without adding conditions for each chain. Because we may add different non EVM chains in future, then we have to add lots of different `if` for each chain to check "nativeCurrency" and other chain related data.